packages/shared/lib/services/sync/data/records.service.ts

```diff
@@ -278,7 +278,7 @@ export async function getDataRecords(
                 END as last_action`),
             'json as record'
         );
-        result = encryptionManager.decryptDataRecords(result, 'record') as unknown as DataRecordWithMetadata[];
+        result = encryptionManager.decryptDataRecordsAndAddCursor(result, 'record') as unknown as DataRecordWithMetadata[];
```

Comment:
as mentioned in the wrong PR :) I am not a huge fan of adding the cursor in the encryption manager. It has nothing to do with encryption and I think the cursor logic should live here in records.service

Reply:
Completely agree, but I didn't want to add another iteration loop in the records service. Could use a generator I suppose

Reply:
or simply make a generic loop where you call both things?
```ts
for (cont res of result) {
  const record = encryptionManager.decryptDataRecord(res)
  record.cursor = addCursor(res);
}
```

Reply:
Yup, and I guess I could also add make the update you just mentioned in Slack too 😅 https://nangohq.slack.com/archives/C067NRBH1LL/p1709626758131279

Reply:
yes indeed that would be a great place; let's just not mix both work but if you do create a generic function I can rewrite the json operation there 👌🏻

Reply:
Updated